wikiclass/feature_lists/wikidatawiki.py

```diff
@@ -14,16 +35,137 @@ class properties:
     DATE_OF_BIRTH = 'P569'
     DATE_OF_DEATH = 'P570'
 
-
 class items:
     """
     Mapping of english descriptions to item idenifiers
     """
     HUMAN = 'Q5'
 
+def _process_all_sources(item):
+
+	list_sources_in_JSON = []
+	
+	try:
+		for property in item.claims:
+			list_of_properties = property
+			for claim in item.claims[list_of_properties]:
+				list_of_claims = claim
+				for i, source in enumerate(list_of_claims.sources):
+					for index_list_1, index_list_2 in source.items(): 
+						sources_in_JSON = index_list_2[0].toJSON()
+						list_sources_in_JSON.append(sources_in_JSON['datavalue']['value'])
+			
+		return list_sources_in_JSON
+	except:
+		return 0 #if the revision does not contain any sources, return 0
+
+def _process_wikimedia_sources(list_all_sources):
+
+	wikimedia_sources = set()
+	result_list = []
+    
+	wdir_path = os.path.dirname(os.path.realpath(__file__)) #get the current working directory path
+	csv_path = os.path.join(wdir_path, 'excluded_qids.csv')
+	
+	#add content from the CSV to set
+	with open(csv_path) as csvfile:
+		readCSV = csv.reader(csvfile, delimiter=',')
+		for line in readCSV:
+			wikimedia_sources.add(int(line[0]))
+	
+	#check revision sources for any Wikimedia project sources
+	for list_all_sources_content in list_all_sources:
+		for wikimedia_sources_content in wikimedia_sources:
+			try:
+				if(list_all_sources_content['numeric-id'] == wikimedia_sources_content): #if a source comes from Wikimedia projects and DBpedia (i.e. DBpedia collects data from Wikipedia), append that source to the list
+					result_list.append(list_all_sources_content)
+					break
+			except:
+				continue
+
+	return result_list
+	
+def _process_unique_sources(item):
+	
```

Comment:
@halfak : The reason of I don't use all_sources in this function is because all_sources only returns the value of sources. For example: "English Wikipedia", "www.forbes.com", "French Wikipedia". On the other hand, if you look at the function _process_unique_sources, it does not only evaluate the value of a source. Rather, it evaluates both the property and value of a source.

So, if there are 2 sources such as, "reference URL": "www.guardian.co.uk" and "stated in": "www.guardian.co.uk", it is going to be counted as 2 unique_sources because their properties are different although their value is similar. IMO, we should look a source as a whole, meaning evaluating both its property and value.
What do you think about this?